libsolutil/Dominator.h

```diff
@@ -376,6 +374,9 @@ class DominatorFinder
 		}
 	}
 
+	// predecessors(w): The set of vertices ``v`` such that (``v``, ``w``) is an edge of the graph.
+	std::vector<std::set<DfsIndex>> m_predecessors;
```

Comment:
Oh, maybe I should have actually extracted that DFS into a separate function like I suggested in https://github.com/ethereum/solidity/pull/14460#discussion_r1676293084. The way I originally wanted to do it was to create a separate struct to hold some precomputed knowledge about the graph that would be from then on treated as const, but could also be stored in a member variable at the end of the algo. It would basically contain `parent`, `predecessor`, `verticesInDFSOrder` and `dfsIndexByVertexId` since all of these things are more like graph metadata and algorithm input rather than its result.

Maybe you should consider doing that later.